connect/api/src/main/java/org/apache/kafka/connect/health/AbstractState.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.connect.health;
+
+/**
+ * {@link AbstractState} provides the current status along with identifier for connect worker and
+ * tasks
+ */
+public abstract class AbstractState {
+
+    private final String state;
+    private final String trace;
+    private final String workerId;
+
+    public AbstractState(String state, String workerId, String trace) {
+        this.state = state;
+        this.workerId = workerId;
+        this.trace = trace;
```

Comment:
Are there any requirements about whether these can be null or empty? What is `trace`? Need JavaDoc since this is part of the API, and also verify the arguments match the requirements.